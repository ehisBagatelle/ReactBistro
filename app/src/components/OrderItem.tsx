import React from "react";
import { TouchableHighlight, View, Text } from "react-native";
import { LineItem } from "../services/OrderService";
import { styles } from "./Components.styles";
import { formatDecimal } from "../helper";

interface Props {
  item: LineItem;
}

const OrderItem: React.FC<Props> = ({ item }) => (
  <View style={styles.rowFront}>
    <View style={styles.item}>
      <Text style={styles.title}>{item.name}</Text>
      <Text style={styles.title}>${formatDecimal(item.price)}</Text>
    </View>
  </View>
);

export default OrderItem;
