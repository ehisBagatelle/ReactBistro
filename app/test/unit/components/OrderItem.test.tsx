import React from 'react';
import renderer from 'react-test-renderer';
import OrderItem from '../../../src/components/OrderItem';

describe('OrderItem component', () => {
  it('renders correctly', () => {
    const mockItem = {
      name: "Test Item",
      category: "Test Category",
      price: 9.99,
      key: "3003"
    };

    const tree = renderer.create(<OrderItem item={mockItem} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
