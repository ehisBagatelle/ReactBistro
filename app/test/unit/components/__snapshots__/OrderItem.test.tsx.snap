// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "height": 50,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "400",
        }
      }
    >
      Test Item
    </Text>
    <Text
      style={
        {
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "400",
        }
      }
    >
      $
      9.99
    </Text>
  </View>
</View>
`;
